test/xbgpu/test_engine.py

```diff
@@ -244,7 +250,11 @@ def generate_expected_beams(
     accum = np.zeros((len(beam_pols), len(batch_indices), n_channels), np.complex64)
     sample = np.empty((N_POLS, COMPLEX), np.int8)
     sample_fp = np.empty(N_POLS, np.complex64)
+
+    beam_params_id = 0
     for batch_id, batch_index in enumerate(batch_indices):
+        if beam_params_change_batch_index == batch_id:
```

Comment:
Should `beam_params_change_batch_index` be renamed to `beam_params_change_batch_id`? It's a little confusing that it's being compared to `batch_id` and not `batch_index` given the name.

Reply:
You're quite right, I was thinking the same and forgot to update before I pushed.